src/main/scala/org/hammerlab/guacamole/assembly/DeBruijnGraph.scala

```diff
@@ -338,22 +339,31 @@ object DeBruijnGraph {
   type Sequence = DeBruijnGraph#Sequence
   type Kmer = DeBruijnGraph#Kmer
 
-  def apply(sequences: Seq[Sequence],
+  def apply(reads: Seq[Read],
             kmerSize: Int,
             minOccurrence: Int = 1,
+            minMeanKmerBaseQuality: Int = 0,
             mergeNodes: Boolean = false): DeBruijnGraph = {
 
     val kmerCounts = mutable.Map.empty[DeBruijnGraph#Kmer, Int]
 
-    sequences
-      .filter(Bases.allStandardBases(_))
-      .foreach(
-        _.sliding(kmerSize)
-          .foreach(seq => {
-            val count = kmerCounts.getOrElse(seq, 0)
-            kmerCounts.update(seq, count + 1)
-          })
-      )
+    reads.foreach(
+      read => {
+        read.sequence
+          .sliding(kmerSize)
+          .zipWithIndex
+          .foreach(
+            { case (seq, index) =>
+              val baseQualities = read.baseQualities.slice(index, index + kmerSize).map(_.toFloat)
+              val meanBaseQuality = mean(baseQualities)
```

Comment:
> these qualities are log-scaled right? does it make sense to arithmetically average them?

Yes and no. Yes, as in it's commonly done for filtering and if most of the bases are 30 and some are 0 this will average ~low 20s and we can filter on that. No, in that there may be better ways to filter. Most of the times it's exactly those cases to filter >3 bases with 'low' base quality.

> follow-up question: is it worth optimizing this to compute/maintain a "streaming" mean, so that you aren't averaging (k-1) of the same qualities at each position that you just did at the previous position?

Yea, I toyed with this - there wasn't a noticeable performance difference on the tests and small runs, but I could imagine on it's own it's much faster. My implementation was a bit messy so I stuck with this, but can invest time in optimizing if this feature stays.